include/unifex/detach_on_cancel.hpp

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include <unifex/get_stop_token.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <atomic>
+#include <exception>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _detach_on_cancel {
+
+template <typename UpstreamSender, typename DownstreamReceiver>
+struct operation_state {
+  struct _receiver;
+
+  struct detached_state;
+
+  struct cancel_callback final {
+    detached_state* state_;
+
+    void operator()() noexcept {
+      auto* op = state_->parentOp_.exchange(nullptr);
+      if (op != nullptr) {
+        // We won the race for cancellation
+        op->state_->stopSource_.request_stop();
+        // ownership is transferred to the detached state
+        (void)op->state_.release();
+        op->callback_.destruct();
+        unifex::set_done(std::move(op->receiver_));
+      }
+    }
+  };
+
+  struct type final {
+    type(UpstreamSender&& s, DownstreamReceiver&& r)
+      : receiver_(std::move(r))
+      , state_(new detached_state(this, (UpstreamSender &&) s)){};
+
+    friend auto tag_invoke(tag_t<unifex::start>, typename operation_state<UpstreamSender, DownstreamReceiver>::type& op) noexcept {
+      auto& childOp = op.state_->childOp_;
+      op.callback_.construct(
+          get_stop_token(op.receiver_), cancel_callback{op.state_.get()});
+      unifex::start(childOp);
+    }
+
+    DownstreamReceiver receiver_;
+    manual_lifetime<typename stop_token_type_t<
+        DownstreamReceiver>::template callback_type<cancel_callback>>
+        callback_;
+    std::unique_ptr<detached_state> state_;
+  };
+};
+
+template <typename UpstreamSender, typename DownstreamReceiver>
+struct operation_state<UpstreamSender, DownstreamReceiver>::_receiver final {
+  auto tryGetOp() noexcept {
+    // if cancellation won, parentOp_ would be null at this point
+    // see line24
```

Comment:
nit: Can we refer to a particular function name instead of line number?
line number references are likely to quickly go out of date as code is modified.